src/Domain/Item/ItemService.ts

```diff
@@ -49,8 +55,138 @@ export class ItemService implements ItemServiceInterface {
     }
   }
 
-  async saveItems(_itemHashes: string[], _userAgent: string, _retrievedItems: Item[]): Promise<SaveItemsResult> {
-    throw new Error('Method not implemented.')
+  async saveItems(dto: SaveItemsDTO): Promise<SaveItemsResult> {
+    const savedItems: Array<Item> = []
+    const conflicts: Array<ItemConflict> = []
+
+    const lastUpdatedTimestamp = this.timer.getTimestampInMicroseconds()
+
+    for (const itemHash of dto.itemHashes) {
+      const existingItem = await this.itemRepository.findByUuidAndUserUuid(itemHash.uuid, dto.userUuid)
+
+      if (!this.itemShouldBeSaved(itemHash, existingItem)) {
+        conflicts.push({
+          serverItem: existingItem,
+          type: 'sync_conflict'
+        })
+
+        continue
+      }
+
+      if(existingItem) {
+        const updatedItem = await this.updateExistingItem(existingItem, itemHash, dto.userAgent)
+        savedItems.push(updatedItem)
+      } else {
+        try {
+          const newItem = await this.saveNewItem(itemHash, dto.userAgent)
+          savedItems.push(newItem)
+        } catch (error) {
```

Comment:
Unused `error` arg. Maybe good to remove or prefix as `_error`

Reply:
Hmm weird that the linter didn't pick this up. Will fix 👍